### PR TITLE
Adding Simulation Experiment - Importance of Compression Feature Number

### PR DESCRIPTION
**Pull request by gwaygenomics**
I thought it would be helpful to perform a simulation experiment and provide a figure that demonstrates how, apart from PCA models, that the compressed feature number is not related to variance explained or importance. In other words, in a VAE model with k = 100, there is no inherent difference in importance between feature 2 and feature 99.

The pull request adds all necessary steps to reproduce. The README contains the experimental design. The figure generated is pasted below:

simulated_feature_number

### CODE REVIEW

File: 11.simulation-feature-number/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+The second group of features (features 5, 6, 7) is simulated to be correlated to a slightly lesser degree (range: 0.85 - 0.90).
+
+| Group | Correlated Features | Correlation Range |
+| :---- | :------------------ | :---------------- |
+| 1 | 1, 2, 3 | 0.90 - 0.95 |
+| 2 | 5, 6, 7 | 0.85 - 0.90 |
+
+## Experimental Design
+
+The experiment is to apply the BioBombe suite of algorithms (PCA, ICA, NMF, DAE, VAE) across 6 different latent dimensionalities (k = 1, 2, 3, 4, 5, 6).
```
Comment (cgreene) [inline 11.simulation-feature-number/README.md]:
```suggestion
For this experiment we apply the BioBombe suite of algorithms (PCA, ICA, NMF, DAE, VAE) across 6 different latent dimensionalities (k = 1, 2, 3, 4, 5, 6).
```

File: 11.simulation-feature-number/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (cgreene) [inline 11.simulation-feature-number/README.md]:
```suggestion
```

File: 11.simulation-feature-number/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+We fit each model and extract the resulting weight matrix.
+We observe the contributions (weights or importance scores) of each input raw feature to each compressed feature.
+
+Our goal is to determine the **top compressed feature number** that represents each simulated feature in both groups.
```
Comment (cgreene) [inline 11.simulation-feature-number/README.md]:
```suggestion
Our goal is to determine the **number** of the compressed feature that best represents each simulated feature in both groups.
```

**Comment by gwaygenomics** [comment]
Two main changes in the recent commits:

* I switched feature number indexing from 0 to 1
* I add a summary figure that may be easier to digest quickly

## Switching feature number indexing:

simulated_feature_number

## Adding summary figure

simulated_feature_number_feature_block_summary

**Comment by cgreene** [comment]
One thing that I notice is that the way models are totaled puts a distinct leftward skew in the summary figure that might be easy to misinterpret, even for models that don't maximize the variability in the earliest components. This is because there are essentially 6 models that could show up with feature 1, 5 for 2, 4, for 3, 3 for 4, 2 for 5, and 1 for 6. Perhaps normalizing by the number of potential models would help with this (as would showing _any_ feature that signal is showing up in) to produce a single summary. Otherwise, it may be best to just show and discuss the row with k=6 where all of these features of the algorithms are apparent.

**Comment by gwaygenomics** [comment]
I decided to plot the k = 6 row as a summary (added in f2da0886c2e6b8d16ceba4495e21367db3640323):

simulated_feature_number_feature_block_summary

It is more intuitive and easier to quickly scan and see the main message. I will go ahead and merge!

**Pull request merged**